File: data/code_pairs/fortran/9.f90
SUBROUTINE SROTG (SA, SB, SC, SS)
    REAL :: SA, SB, SC, SS
    REAL :: U, V, R

    IF (ABS(SA) > ABS(SB)) THEN
        U = SA + SA
        V = SB / U
        R = SQRT(0.25E0 + V**2) * U
        SC = SA / R
        SS = V * (SC + SC)
        SB = SS
        SA = R
    ELSE IF (SB /= 0.0E0) THEN
        U = SB + SB
        V = SA / U
        SA = SQRT(0.25E0 + V**2) * U
        SS = SB / SA
        SC = V * (SS + SS)
        IF (SC /= 0.0E0) THEN
            SB = 1.0E0 / SC
        ELSE
            SB = 1.0E0
        END IF
    ELSE
        SC = 1.0E0
        SS = 0.0E0
    END IF
END SUBROUTINE SROTG

File: data/code_pairs/cpp/9.cpp
#include <cmath>

void SROTG(double &SA, double &SB, double &SC, double &SS) {
    double U, V, R;
    
    if (std::abs(SA) > std::abs(SB)) {
        U = SA + SA;
        V = SB / U;
        R = std::sqrt(0.25 + V*V) * U;
        SC = SA / R;
        SS = V * (SC + SC);
        SB = SS;
        SA = R;
    } else if (SB != 0.0) {
        U = SB + SB;
        V = SA / U;
        R = std::sqrt(0.25 + V*V) * U;
        SA = R;
        SS = SB / SA;
        SC = V * (SS + SS);
        if (SC != 0.0) {
            SB = 1.0 / SC;
        } else {
            SB = 1.0;
        }
    } else {
        SC = 1.0;
        SS = 0.0;
    }
}